src/MarkovChainMonteCarlo.jl

```diff
@@ -3,307 +3,587 @@ module MarkovChainMonteCarlo
+    em::Emulator{FT}, 
+    obs_sample::Vector{FT}
+) where {FT <: AbstractFloat}
+    # recall predict() written to return multiple `N_samples`: expects input to be a Matrix
+    # with `N_samples` columns. Returned g is likewise a Matrix, and g_cov is a Vector of 
+    # `N_samples` covariance matrices. For MH, N_samples is always 1, so we have to 
+    # reshape()/re-cast input/output; simpler to do here than add a predict() method.
+    return AdvancedMH.DensityModel(
+        function (θ) 
+            # θ: model params we evaluate at; in original coords.
+            # transform_to_real = false means g, g_cov, obs_sample are in decorrelated coords.
+            g, g_cov = Emulators.predict(em, reshape(θ,:,1), transform_to_real=false)
+            return logpdf(MvNormal(obs_sample, g_cov[1]), vec(g)) + get_logpdf(prior, θ)
+       end
+    )
+end
 
-abstract type AbstractMCMCAlgo end
-struct RandomWalkMetropolis <: AbstractMCMCAlgo end
+# ------------------------------------------------------------------------------------------
+# Extend proposal distribution object to allow for tunable stepsize
 
 """
-    MCMC{FT<:AbstractFloat, IT<:Int}
+    VariableStepProposal
 
-Structure to organize MCMC parameters and data
+`AdvancedMH.Proposal` object responsible for generating the random walk steps used for new 
+parameter proposals in the Metropolis-Hastings algorithm. Adds a separately adjustable 
+`stepsize` parameter to the implementation from 
+[AdvancedMH](https://github.com/TuringLang/AdvancedMH.jl).
+
+This is allowed to be mutable, in order to dynamically change the `stepsize`. Since no
+sampler state gets stored here, this choice is made out of convenience, not necessity.
 
 # Fields
 $(DocStringExtensions.FIELDS)
 """
-struct MCMC{FT<:AbstractFloat, IT<:Int}
-    "a single sample from the observations. Can e.g. be picked from an Obs struct using get_obs_sample"
-    obs_sample::Vector{FT}
-    "covariance of the observational noise"
-    obs_noise_cov::Array{FT, 2}
-    "array of length N_parameters with the parameters' prior distributions"
-    prior::ParameterDistribution
-    "MCMC step size"
-    step::Array{FT}
-    "Number of MCMC steps that are considered burnin"
-    burnin::IT
-    "the current parameters"
-    param::Vector{FT}
-    "Array of accepted MCMC parameter samples. The histogram of these samples gives an approximation of the posterior distribution of the parameters. param_dim x n_samples"
-    posterior::Array{FT, 2}
-    "the (current) value of the logarithm of the posterior (= log_likelihood + log_prior of the current parameters)"
-    log_posterior::Array{Union{FT, Nothing}}
-    "iteration/step of the MCMC"
-    iter::Array{IT}
-    "number of accepted proposals"
-    accept::Array{IT}
-    "MCMC algorithm to use - currently implemented: 'rmw' (random walk Metropolis)"
-    algtype::String
+mutable struct VariableStepProposal{issymmetric, P, FT<:AbstractFloat} <: AdvancedMH.Proposal{P}
```

Comment:
Good point, I hadn't thought that this might create confusion with variable stepsize methods. Presumably the SVD decorrelation makes the emulated likelihood approximately isotropic, so implementing a variable step method wouldn't be worth it.

I agree that immutability shouldn't be abandoned lightly, but since the stepper itself is a field in `MCMCWrapper`, we'd have to recreate that entire object every time we wanted to change the step size. There are ways to make this less clunky, like [Setfield.jl](https://github.com/jw3126/Setfield.jl), but what's really being asked for in something like `optimize_stepsize!` is mutability.

What do you think of the following instead: we take `stepsize` out of the PriorProposalMHSampler/ScaledPriorProposal objects entirely and make it an optional argument on the sample() function:
- The stepper/sampler objects are now immutable;
- We're defining new sample() methods anyway, to use `MCMCWrapper` object;
- `stepsize` as we've defined it is a relative, not absolute quantity, since it scales the covariance of the prior that the stepper uses, so tying it to the sampling run might be a more logical presentation.

Reply:
The above proposal is implemented in 9673925. In addition to removing the use of a mutable struct, it also eliminates the need for a chunk of AbstractMCMC/AdvancedMH boilerplate.